pycross/private/tools/pdm_translator.py

```diff
@@ -209,13 +213,15 @@ def translate(
 
         dependencies = {Requirement(dep) for dep in lock_pkg.get("dependencies", [])}
         files = {parse_file_info(f) for f in lock_pkg.get("files", [])}
+        is_local = "path" in lock_pkg and "files" not in lock_pkg
```

Comment:
Maybe it makes sense to just ignore these "local" packages and output a warning? As you say, they can already be accessed via Bazel.

Reply:
I don't think we can just ignore the local packages outright. Under a monorepo-style layout, local packages form the roots of the dependency graph; dropping them would make for an empty dependency set as seen by the lock resolver ([src](https://github.com/jvolkman/rules_pycross/blob/3d4b8a9d932207ba09ca56858f3352dd1f2557a4/pycross/private/tools/raw_lock_resolver.py#L406)).

But maybe what we can do is to expand the local packages to their deps such that the local packages do not appear beyond the scope of the PDM translator. That is, we still have to handle them somewhat when translating the PDM lockfile, but we don't have to pollute the lock model or anything downstream of it. What do you think?

Reply:
Sorry @plobsing, have been on vacation for a bit.

I think I understand, and it makes sense to me to "promote" the dependencies of each local package into the set of pinned packages we start from, keeping all of the details hidden in the PDM translation step. :+1: 

It looks like you're handling that below, correct?

FWIW I'm also working on native support for extras ([here](https://github.com/jvolkman/rules_pycross/tree/dev/extra-support)). If I remember correctly, the "merge" stuff should go away with this.

Reply:
> Sorry @plobsing, have been on vacation for a bit.

I'm just happy for any time you can spare for this.
 
> I think I understand, and it makes sense to me to "promote" the dependencies of each local package into the set of pinned packages we start from, keeping all of the details hidden in the PDM translation step. 👍
> 
> It looks like you're handling that below, correct?

Correct, that's what's going on with the loop starting at line 281.

> FWIW I'm also working on native support for extras ([here](https://github.com/jvolkman/rules_pycross/tree/dev/extra-support)). If I remember correctly, the "merge" stuff should go away with this.

Cool. I tried to play nice with the merge logic, but TBH I never fully grokked it. Will be nice to see that logic go. Is there anything that affects this PR and how do you want to handle that, given the extras support isn't merged to `main` yet?

Reply:
No, I don't think so. I'll handle that merge.